KubeArmor/core/kubeArmor.go

```diff
@@ -609,23 +609,23 @@ func KubeArmor() {
 		dm.Logger.Print("Started to monitor host security policies")
 	}
 
-	policyService := &policy.ServiceServer{}
-
-	if enableContainerPolicy {
-		policyService.UpdateContainerPolicy = dm.ParseAndUpdateContainerSecurityPolicy
-		dm.Logger.Print("Started to monitor container security policies on gRPC")
-	}
-
-	if !cfg.GlobalCfg.K8sEnv && cfg.GlobalCfg.HostPolicy {
-		policyService.UpdateHostPolicy = dm.ParseAndUpdateHostSecurityPolicy
-		dm.Node.PolicyEnabled = tp.KubeArmorPolicyEnabled
+	if !dm.K8sEnabled && (enableContainerPolicy || cfg.GlobalCfg.HostPolicy) {
```

Comment:
do we need the & condition (enableContainerPolicy || cfg.GlobalCfg.HostPolicy) when we are already checking it inside the if condition?

Reply:
Yes, we only register the service if either of them are enabled.